src/misc_functions/context.ts

```diff
@@ -0,0 +1,20 @@
+export interface ContextPath<T> {
+    path: string[];
+    data: (args: string[]) => T;
+}
+
+export function resolvePaths<T>(paths: Array<ContextPath<T>>, argPath: string[], args: string[]): T | undefined {
+    for (const path of paths) {
+        if (arreq(argPath, path.path)) {
+            return path.data(args);
+        }
+    }
+    return undefined;
+}
+
+function arreq(arr1: string[], arr2: string[]) {
```

Comment:
Can you call this `arraysEqual`, and maybe make a note that it only checks for reference equality.